Zotero/Scenes/Detail/PDF/PDFCoordinator.swift

```diff
@@ -285,6 +296,134 @@ extension PDFCoordinator: PdfReaderCoordinatorDelegate {
+                        coordinator.share(item: image, sourceView: .view(sender, nil))
+                    } else {
+                        (self as Coordinator).share(item: image, sourceView: .view(sender, nil))
+                    }
+                }
+                action.accessibilityLabel = L10n.Accessibility.Pdf.shareAnnotationImage + " " + title
+                action.isAccessibilityElement = true
+                elementProvider([action])
+            } onFailure: { (error: Error) in
+                DDLogError("PDFCoordinator: can't render annotation image - \(error)")
+                elementProvider([])
+            }
+            .disposed(by: disposeBag)
+        }
+    }
+        
+    func shareAnnotationMenu(
+        state: PDFReaderState,
+        annotation: Annotation,
+        sender: UIButton
+    ) -> UIMenu? {
+        var children: [UIMenuElement] = []
+        if annotation.type == .image,
+           let boundingBoxConverter = self.navigationController.viewControllers.last as? AnnotationBoundingBoxConverter
+        {
+            var shareImageMenuChildren: [UIMenuElement] = []
+            for (scale, title) in [
+                (300.0 / 72.0, L10n.Pdf.AnnotationShare.Image.medium),
+                (600.0 / 72.0, L10n.Pdf.AnnotationShare.Image.large)
+            ] {
+                let menuElement = deferredShareImageMenuElement(state: state, annotation: annotation, sender: sender, boundingBoxConverter: boundingBoxConverter, scale: scale, title: title)
+                shareImageMenuChildren.append(menuElement)
+            }
+            let shareImageMenu = UIMenu(title: L10n.Pdf.AnnotationShare.Image.share, options: [.displayInline], children: shareImageMenuChildren)
+            children.append(shareImageMenu)
+        }
+        guard !children.isEmpty else { return nil }
+        return UIMenu(children: children)
+    }
+    
+    func shareAnnotationImage(
+        state: PDFReaderState,
+        annotation: Annotation,
+        scale: CGFloat = 1.0,
+        sender: UIButton
+    ) {
+        guard annotation.type == .image,
+              let pdfReaderViewController = navigationController.viewControllers.last as? PDFReaderViewController
+        else { return }
+        let annotationPreviewController = controllers.annotationPreviewController
+        let pageIndex: PageIndex = UInt(annotation.page)
+        let rect = annotation.boundingBox(boundingBoxConverter: pdfReaderViewController)
+        var size = rect.size
+        size.width *= scale
+        size.height *= scale
+        annotationPreviewController.render(
+            document: state.document,
+            page: pageIndex,
+            rect: rect,
+            imageSize: size,
+            imageScale: 1.0,
+            key: annotation.key,
+            parentKey: state.key,
+            libraryId: state.library.id
+        )
+        .observe(on: MainScheduler.instance)
+        .subscribe { [weak self] (image: UIImage) in
```

Comment:
Don't need to specify `image: UIImage`, I don't remember whether we added any rule in SwiftLint to this, so we can continue this discussion there.

Reply:
Same as above. Check https://github.com/zotero/zotero-ios/pull/693/commits/6a6701f345c724855358ba48677131e1c4236d21